sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java

```diff
@@ -570,20 +631,158 @@ public ServiceBusReceiverClientBuilder subscriptionName(String subscriptionName)
+     * messages from Service Bus.
+     */
+    @ServiceClientBuilder(serviceClients = {ServiceBusReceiverClient.class, ServiceBusReceiverAsyncClient.class})
+    public final class ServiceBusReceiverClientBuilder {
+        private int prefetchCount = DEFAULT_PREFETCH_COUNT;
+        private String queueName;
+        private ReceiveMode receiveMode = ReceiveMode.PEEK_LOCK;
+        private String subscriptionName;
+        private String topicName;
+
+        private ServiceBusReceiverClientBuilder() {
+        }
+
+        /**
+         * Sets the prefetch count of the receiver. Prefetch speeds up the message flow by aiming to have a message
+         * readily available for local retrieval when and before the application asks for one using {@link
+         * ServiceBusReceiverAsyncClient#receive()}. Setting a non-zero value will prefetch that number of messages.
+         * Setting the value to zero turns prefetch off. For both {@link ReceiveMode#PEEK_LOCK PEEK_LOCK} and {@link
+         * ReceiveMode#RECEIVE_AND_DELETE RECEIVE_AND_DELETE} modes the default value is 1.
+         *
+         * @param prefetchCount The prefetch count.
          *
          * @return The modified {@link ServiceBusReceiverClientBuilder} object.
          */
-        public ServiceBusReceiverClientBuilder sessionId(String sessionId) {
-            this.sessionId = sessionId;
+        public ServiceBusReceiverClientBuilder prefetchCount(int prefetchCount) {
+            this.prefetchCount = prefetchCount;
+            return this;
+        }
+
+        /**
+         * Sets the name of the queue to create a receiver for.
+         *
+         * @param queueName Name of the queue.
+         *
+         * @return The modified {@link ServiceBusReceiverClientBuilder} object.
+         */
+        public ServiceBusReceiverClientBuilder queueName(String queueName) {
+            this.queueName = queueName;
+            return this;
+        }
+
+        /**
+         * Sets the receive mode for the receiver.
+         *
+         * @param receiveMode Mode for receiving messages.
+         *
+         * @return The modified {@link ServiceBusReceiverClientBuilder} object.
+         */
+        public ServiceBusReceiverClientBuilder receiveMode(ReceiveMode receiveMode) {
+            this.receiveMode = receiveMode;
+            return this;
+        }
+
+        /**
+         * Sets the name of the subscription in the topic to listen to.
+         *
+         * @param subscriptionName Name of the subscription.
+         *
+         * @return The modified {@link ServiceBusReceiverClientBuilder} object.
+         * @see #topicName A topic name should be set as well.
+         */
+        public ServiceBusReceiverClientBuilder subscriptionName(String subscriptionName) {
+            this.subscriptionName = subscriptionName;
+            return this;
+        }
+
+        /**
+         * Sets the name of the topic.
+         *
+         * @param topicName Name of the topic.
```

Comment:
To avoid using doing mistake and not setting session id :  `Sets the name of the subscription in the topic to listen to. You must also set {@code subscriptionName}`.